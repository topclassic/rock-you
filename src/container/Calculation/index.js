import React, { Component } from "react";
import styled from "styled-components";
import { Card, Button, Form, Input, Icon, Tooltip, notification } from "antd";
import { validateNumber } from "../../utils/validate";
import { normalizeDigit } from "../../utils/normalize";
import CalSmallestAmount from "../../utils/calSmallestAmount";

const Circle = styled.span`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 16px;
  width: 16px;
  background-color: #e2e2e2;
  border-radius: 50%;
`;

const GroupRight = styled.div`
  display: flex;
  justify-content: flex-end;
  align-items: center;
`;

const text = "Calculate smallest quantity of bill and coins";
export class CalContainer extends Component {
  static defaultProps = {
    form: {}
  };
  handleSubmit = e => {
    e.preventDefault();
    this.props.form.validateFields((err, values) => {
      if (!err) {
        if (values.amount) {
          notification.success({
            duration: 0,
            message: `Calculate amount with ${values.amount}`,
            description: CalSmallestAmount(values.amount)
          });
        }
      }
    });
  };
  render() {
    const { getFieldDecorator } = this.props.form;
    return (
      <Form onSubmit={this.handleSubmit}>
        <Card
          style={{
            display: "flex",
            justifyContent: "center",
            alignItems: "center",
            flexDirection: "column"
          }}
        >
          <Form.Item>
            {getFieldDecorator("amount", {
              normalize: value => normalizeDigit(value),
              rules: [
                {
                  validator: (rule, value, callback) =>
                    validateNumber(rule, value, callback),
                  required: true
                }
              ]
            })(
              <Input
                style={{ width: 250 }}
                suffix={
                  <Tooltip placement="right" title={text}>
                    <Circle>
                      <Icon style={{ fontSize: 10 }} type="question" />
                    </Circle>
                  </Tooltip>
                }
                placeholder="number"
              />
            )}
          </Form.Item>
          <GroupRight>
            <Button type="primary" htmlType="submit">
              Calculate
            </Button>
          </GroupRight>
        </Card>
      </Form>
    );
  }
}

const CalContainerForm = Form.create({ name: "calContainer" })(CalContainer);

export default CalContainerForm;
