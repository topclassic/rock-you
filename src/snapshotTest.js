
import renderer from 'react-test-renderer'

const snapshotTest = (testCase, element) => {
  it(testCase, () => {
    const wrapper = renderer.create(element).toJSON()
    expect(wrapper).toMatchSnapshot()
  })
}

export default snapshotTest