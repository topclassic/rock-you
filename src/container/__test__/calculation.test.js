import React from 'react';
import { shallow } from 'enzyme';
import Calculation from '../Calculation';
import snapshotTest from "../../snapshotTest";

describe("Calculation", () => {
  it("test render component correctly", () => {
    const wrapper = shallow(<Calculation />);
    expect(wrapper).toHaveLength(1);
  });
});

describe("Calculation testing", () => {
  snapshotTest("render Calculation", <Calculation />);
});
