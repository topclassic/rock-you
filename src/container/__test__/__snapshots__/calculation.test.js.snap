// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculation testing render Calculation 1`] = `
<form
  className="ant-form ant-form-horizontal"
  onSubmit={[Function]}
>
  <div
    className="ant-card ant-card-bordered"
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <div
      className="ant-card-body"
      style={Object {}}
    >
      <div
        className="ant-row ant-form-item"
      >
        <div
          className="ant-col ant-form-item-control-wrapper"
        >
          <div
            className="ant-form-item-control"
          >
            <span
              className="ant-form-item-children"
            >
              <span
                className="ant-input-affix-wrapper"
                style={
                  Object {
                    "width": 250,
                  }
                }
              >
                <input
                  className="ant-input"
                  data-__field={
                    Object {
                      "name": "amount",
                    }
                  }
                  data-__meta={
                    Object {
                      "name": "amount",
                      "normalize": [Function],
                      "originalProps": Object {
                        "disabled": false,
                        "placeholder": "number",
                        "style": Object {
                          "width": 250,
                        },
                        "suffix": <Tooltip
                          arrowPointAtCenter={false}
                          autoAdjustOverflow={true}
                          mouseEnterDelay={0.1}
                          mouseLeaveDelay={0.1}
                          placement="right"
                          title="Calculate smallest quantity of bill and coins"
                          transitionName="zoom-big-fast"
                        >
                          <ForwardRef>
                            <Icon
                              style={
                                Object {
                                  "fontSize": 10,
                                }
                              }
                              type="question"
                            />
                          </ForwardRef>
                        </Tooltip>,
                        "type": "text",
                      },
                      "ref": null,
                      "rules": Array [
                        Object {
                          "required": true,
                          "validator": [Function],
                        },
                      ],
                      "trigger": "onChange",
                      "validate": Array [
                        Object {
                          "rules": Array [
                            Object {
                              "required": true,
                              "validator": [Function],
                            },
                          ],
                          "trigger": Array [
                            "onChange",
                          ],
                        },
                      ],
                      "valuePropName": "value",
                    }
                  }
                  disabled={false}
                  id="calContainer_amount"
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  placeholder="number"
                  style={null}
                  type="text"
                  value=""
                />
                <span
                  className="ant-input-suffix"
                >
                  <span
                    className="sc-bdVaJa fTCfLx"
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                  >
                    <i
                      aria-label="icon: question"
                      className="anticon anticon-question"
                      style={
                        Object {
                          "fontSize": 10,
                        }
                      }
                    >
                      <svg
                        aria-hidden="true"
                        className=""
                        data-icon="question"
                        fill="currentColor"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M764 280.9c-14-30.6-33.9-58.1-59.3-81.6C653.1 151.4 584.6 125 512 125s-141.1 26.4-192.7 74.2c-25.4 23.6-45.3 51-59.3 81.7-14.6 32-22 65.9-22 100.9v27c0 6.2 5 11.2 11.2 11.2h54c6.2 0 11.2-5 11.2-11.2v-27c0-99.5 88.6-180.4 197.6-180.4s197.6 80.9 197.6 180.4c0 40.8-14.5 79.2-42 111.2-27.2 31.7-65.6 54.4-108.1 64-24.3 5.5-46.2 19.2-61.7 38.8a110.85 110.85 0 0 0-23.9 68.6v31.4c0 6.2 5 11.2 11.2 11.2h54c6.2 0 11.2-5 11.2-11.2v-31.4c0-15.7 10.9-29.5 26-32.9 58.4-13.2 111.4-44.7 149.3-88.7 19.1-22.3 34-47.1 44.3-74 10.7-27.9 16.1-57.2 16.1-87 0-35-7.4-69-22-100.9zM512 787c-30.9 0-56 25.1-56 56s25.1 56 56 56 56-25.1 56-56-25.1-56-56-56z"
                        />
                      </svg>
                    </i>
                  </span>
                </span>
              </span>
            </span>
          </div>
        </div>
      </div>
      <div
        className="sc-bwzfXH faZFfx"
      >
        <button
          className="ant-btn ant-btn-primary"
          onClick={[Function]}
          type="submit"
        >
          <span>
            Calculate
          </span>
        </button>
      </div>
    </div>
  </div>
</form>
`;
